run_bin_test.go

```diff
@@ -578,6 +780,67 @@ func TestCoverageCollector_RunBinary(t *testing.T) {
 	}
 }
 
+func stdinPipePreFuncCovCollectorOption() CoverageCollectorOption {
+	f := PreCmdFunc(func(cmd *exec.Cmd) error {
+		writer, _ := cmd.StdinPipe()
+		_, err := writer.Write([]byte("Hello from prefunc\n"))
+		writer.Close()
+		return err
+	})
+	return PreExec(f)
+}
+
+func errPreCmdFuncCovCollectorOption() CoverageCollectorOption {
+	return PreExec(errPreCmdFunc())
+}
+
+func errPostCmdFuncCovCollectorOption() CoverageCollectorOption {
+	return PostExec(errPostCmdFunc())
+}
+
+func nilPreCmdFunc() CoverageCollectorOption {
+	f := PreCmdFunc(func(cmd *exec.Cmd) error {
+		return nil
+	})
+	return PreExec(f)
+}
+
+func printToBufferPreFunc(buffer *string) PreCmdFunc {
+	f := PreCmdFunc(func(cmd *exec.Cmd) error {
+		*buffer += "Hello from prefunc\n"
```

Comment:
I try to minimize magic strings like this. It wasn't clear to me why in the test code you were asserting "Hello from prefunc" as I didn't see it declared anywhere. I would change the signature of this function to take in a string, pass in that string in the test, and then assert that string in `want`.